hash.go

```diff
@@ -46,6 +46,34 @@ func (db *RoseDB) HGet(key, field []byte) ([]byte, error) {
 	return val, err
 }
 
+// HMGet Returns the values associated with the specified fields in the hash stored at key.
+// HMGet For every field that does not exist in the hash, a nil value is returned.
+// Because non-existing keys are treated as empty hashes, running HMGET against a non-existing key will return a list of nil values.
+func (db *RoseDB) HMGet(key []byte, field ...[]byte) (vals [][]byte, err error) {
+	db.hashIndex.mu.RLock()
+	defer db.hashIndex.mu.RUnlock()
+
+	length := len(field)
+	if db.hashIndex.trees[string(key)] == nil { // key not exist
+		for i := 0; i < length; i++ {
+			vals = append(vals, nil)
+		}
+		return vals, nil
+	}
+	// key exist
+	db.hashIndex.idxTree = db.hashIndex.trees[string(key)]
+
+	for _, v := range field {
+		val, err := db.getVal(v, Hash)
+		if err == ErrKeyNotFound {
```

Comment:
return `err` if err is not ErrKeyNotFound

Reply:
according to https://redis.io/commands/hmget/
if field does not exist, there should keep nil at that position, cannot return directly.
![image](https://user-images.githubusercontent.com/38450639/167287082-ee6310f9-e21a-4e45-8a57-0a2a57172251.png)